### ISSUE TITLE
header-title.php

### ISSUE THREAD

**Issue by j3hn** [joshuaiz/plate#18]
Not an issue just interested in why abstracting the headers of each template into the header-template is done in this case. I can't see the benefits in this case, please what am I missing here? Loving the template, best starter I've found.

**Comment by joshuaiz** [comment]
Hey @jpbash,

Thanks for the support!

Not exactly sure what you mean but the main reason to have `header.php` in a different file is in case you wanted to have a custom header for a particular page.

We have `header.php` and `header-full.php` in Plate to work with CSS Grid. The default `header.php` uses the `grid-aside` class which has a sidebar whereas `header-full.php` uses `grid-full` and is full-width.

Thus if you wanted to use a full-width grid layout, in your page template you could make a call like this:

`<?php get_header('full'); ?>`

You could even create your own custom header with a `header-custom.php` file and then call it:

`<?php get_header('custom'); ?>`

This could come in handy if you wanted to have a hero in the header or maybe a secondary nav but only on some pages.

Let me know if that still doesn't answer your question.

J

**Comment by j3hn** [comment]
Sorry I wasn't very clear in my question. I was referring to the header-title.php file. You abstract away page template titles ( archive.php, search.php, 404.php ) into this file, with the titles only being one line I didn't understand why you abstracted them into the header-title.php file. I was wondering the purpose and though I'd ask.

**Comment by joshuaiz** [comment]
The convention of adding a title to the header file follows the WordPress template hierarchy. 

So if I have say `header-custom.php` and in any page template I can substitute the normal header call which is:

`<?php get_header(); ?>`

with:

`<?php get_header('custom'); ?>`

Then WordPress will look for a file called `header-custom.php` (with the dash). Whatever you have in the parentheses should match what you have after the dash in your file name. This works similar to WordPress page templates or custom post type files.

A similar example is if you had a custom post type called `movie`. By default, WordPress will use the `single.php` for any single posts but if you wanted to create a custom single post template for your `movie` post type, you would name it `single-movie.php` and WordPress will automatically use that for single posts of type `movie`. 

WordPress will search through your theme directory to find the first matching file in the hierarchy. If you don't have a `single-movie.php` file, it will use `single.php` and if you don't have a `single.php` file (which is technically not required for a working theme), it will use `index.php`.

You can see more how this works here: https://wphierarchy.com

...although they don't cover the `header-[whatever].php` case but it works the same way.

In short, the `header-[whatever].php` convention allows you to create as many custom header templates as you want and call them in your theme wherever you want.

Does that make sense?

**Comment by j3hn** [comment]
I'm so sorry for making you write all that. I understand the template hierarchy fine. I've used the wrong terms to ask what I hoped was a simple answer. It's not even an issue. Please don't feel obliged to answer but I'll try one last time for retribution. 

You use <?php get_template_part( 'templates/header', 'title'); ?> in a number of templates to keep the titles in a single templates/header-title.php file. I couldn't see the benefit of the extra code required to do so compared to just having the title code in the template. I was wondering if it was a personal preference or there were benefits to this setup I can't see?

**Comment by joshuaiz** [comment]
No worries - sorry I got it now :)

There really is no benefit to having all of the title code with conditionals in one file. You're right, since it is so little code, it probably isn't worth splitting out into the template part as that makes it harder to edit for each template.

I think it makes sense to remove that template part and just add the title code to each page so I'll sort that out for the next update.

**Comment by j3hn** [comment]
Ahh great we're on the same page then. Sorry for the mess around and thanks for the discussion.

**Comment by joshuaiz** [comment]
Commit  1d90e32c7b2034e26e066be743d41f2b2353c891 closes #18
